GoogleForCreators/web-stories-wp
# Add missing title dialog

## Summary

This adds a "missing title" dialog that appears when trying to publish without a title.

This is what it looks like:
![Screen Shot 2020-08-31 at 22 45 23](https://user-images.githubusercontent.com/637548/91789166-0c4e1e00-ebdc-11ea-9994-bf06647aa3d0.png)

And this is how the flow works:
![title_adding](https://user-images.githubusercontent.com/637548/91789172-14a65900-ebdc-11ea-95f3-46b5b035fa03.gif)

## Relevant Technical Choices

* Added a necessary context to the header to be able to focus the title input from the publish button.

## To-do

- [x] Get the correct link to guidelines. #4184 links to a [Google Drive folder](https://drive.google.com/drive/folders/1dqPtjNTYN7OxTngWAYhy8zn8X3EuRTGJ), but it's protected, so not publicly accessible.
- [x] Check design with @samitron7 
- [x] Test functionality with unit tests
- [ ] Test design with screenshots with karma tests

## User-facing changes

* If you try to publish without a title, you get a dialog.

## Testing Instructions

1. Create a story without a title
1. Press the publish button and observe a dialog warning about missing title
1. Press "Add a title" and observe that after the dialog closes, the story will still be in _draft_ state and the keyboard focus will move to the title input

---



Fixes #4184
